### PR TITLE
Fixing npm tasks to get location properly

### PR DESCRIPTION
**Pull request by jahquin-msft**

### CODE REVIEW

File: Tasks/Common/npm-common/Authentication.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import * as tl from "vsts-task-lib/task";
+
+export interface IPackageSource {
+    feedName: string;
+    feedUri: string;
+    isInternal: boolean;
+}
+
+export class NuGetAuthInfo {
```
Comment (aldoms) [inline Tasks/Common/npm-common/Authentication.ts]: NpmAuthInfo?

File: Tasks/Common/npm-common/Authentication.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    constructor(
+        public uriPrefixes: string[],
+        public accessToken: string) {
+    }
+}
+
+export class NuGetExtendedAuthInfo {
```
Comment (aldoms) [inline Tasks/Common/npm-common/Authentication.ts]: Same here

File: Tasks/Common/npm-common/Authentication.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    constructor(
+        public internalAuthInfo: InternalAuthInfo,
+        public externalAuthInfo?: ExternalAuthInfo[]) {
+    }
+}
+
+export class InternalAuthInfo
```
Comment (aldoms) [inline Tasks/Common/npm-common/Authentication.ts]: Is this being used?

File: Tasks/Common/npm-common/Authentication.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+export enum ExternalAuthType
+{
+    Token, 
+    UsernamePassword,
+    ApiKey
+}
+
+export function getSystemAccessToken(): string {
```
Comment (aldoms) [inline Tasks/Common/npm-common/Authentication.ts]: This should already be available from the util.ts [file](https://github.com/Microsoft/vsts-tasks/blob/master/Tasks/Common/npm-common/util.ts#L143).

File: Tasks/Common/npm-common/LocationHelpers.ts [hunk 5]
```diff
@@ -0,0 +1,208 @@
+import * as Q from "q";
+import * as url from "url";
+import * as util from "util";
+import * as vstsWebApi from "vso-node-api/WebApi";
+import * as tl from "vsts-task-lib/task";
+
+import * as auth from "./Authentication";
+import * as locationApi from "./LocationApi";
+
+export const NUGET_ORG_V2_URL: string = "https://www.nuget.org/api/v2/";
```
Comment (aldoms) [inline Tasks/Common/npm-common/LocationHelpers.ts]: These two aren't being used.

File: Tasks/Common/npm-common/Authentication.ts (hunk 1 above)
Comment (jahquin-msft) [reply Tasks/Common/npm-common/Authentication.ts]: Removed file altogether.

File: Tasks/Common/npm-common/Authentication.ts (hunk 2 above)
Comment (jahquin-msft) [reply Tasks/Common/npm-common/Authentication.ts]: Removed file altogether.

File: Tasks/Common/npm-common/Authentication.ts (hunk 3 above)
Comment (jahquin-msft) [reply Tasks/Common/npm-common/Authentication.ts]: Removed file altogether.

File: Tasks/Common/npm-common/Authentication.ts (hunk 4 above)
Comment (jahquin-msft) [reply Tasks/Common/npm-common/Authentication.ts]: Fixed! :)

File: Tasks/Common/npm-common/LocationHelpers.ts (hunk 5 above)
Comment (jahquin-msft) [reply Tasks/Common/npm-common/LocationHelpers.ts]: Removed.

File: Tasks/Common/npm-common/LocationHelpers.ts [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+                            .then((targetConnectionData) => {
+                                tl.debug("successfully loaded target service location data");
+                                return targetConnectionData;
+                            });
+                    });
+            }
+        });
+}
+
+export function getNuGetConnectionData(serviceUri: string, accessToken: string): Q.Promise<locationApi.ConnectionData> {
```
Comment (aldoms) [inline Tasks/Common/npm-common/LocationHelpers.ts]: I don't think this is being used either.

**Comment by jahquin-msft** [comment]
Windows build failed due to an unrelated test failure for the AzurePowerShell task.

**Pull request merged**